src/pspm_display.m

```diff
@@ -765,7 +796,35 @@ function ed_y_max_CreateFcn(hObject, eventdata, handles)
             
             if strcmp(handles.prop.event,'extra')
                 set(hbase,'BaseValue',base(2),'Visible','off');
-            else set(hbase,'BaseValue',base(1),'Visible','off');
+            elseif strcmp(handles.prop.event,'integrated')
+                set(hbase,'BaseValue',base(1),'Visible','off');
+            end
+        elseif not(isempty(events))
+            
+            events=round(events*sr.wave);
+            
+            EVENTS = zeros(size(wave));
+                        
+            if strcmp(handles.prop.event,'extra')
+                EVENTS(events,1)=min(wave)-.5;
+            elseif strcmp(handles.prop.event,'integrated')
+                temp=wave(events,1);
+                median_non_nan_vals = median(temp,'omitnan');
+                EVENTS(events,1)=temp;
+                EVENTS(isnan(EVENTS))=median_non_nan_vals;
+            end
+            
+            EVENTS(EVENTS==0) = NaN;
```

Comment:
Hey, just checking i understand this bit

L806, EVENTS is set to a zero matrix
L808-L814, values are assigned to the EVENTS according to the plotting strategy and main data channel values, also setting NaN to the median value of the wave
L817 EVENTS has it's zero's set to NaN.

Does this mean there's an assumption that the wave values will never be 0, otherwise the events value would be overwritten to NaN?

```matlab
            events=round(events*sr.wave);

            EVENTS = nan(size(wave));

            if strcmp(handles.prop.event,'extra')
                EVENTS(events,1)=min(wave)-.5;
            elseif strcmp(handles.prop.event,'integrated')
                temp=wave(events,1);
                temp(isnan(temp)) = median(temp,'omitnan');
                EVENTS(events,1)=temp;
            end

```

Does this achieve the same without requiring that assumption?
1. initate EVENTS as NaN.
2. set EVENTS to the wave data after filling wave NaNs with median vals.
3. no need to set EVENTS = 0 to NaN

Reply:
That's true I haven't noticed that, I've just inspired myself from the `marker` case, I'll check that